### PR TITLE
chore(core): Update core/tsconfig.json to include all typescript files

### PR DESCRIPTION
**Pull request by christopherthielen**
Make things compile -- export interfaces as necessary

### CODE REVIEW

**Comment by christopherthielen** [comment]
Some Typescript files were reachable only through a `.js` file.  
I switched to a glob `includes`, then exported and renamed all interfaces as necessary.

File: app/scripts/modules/core/src/delivery/create/TemplateSelector.tsx [hunk 1]
```diff
@@ -25,7 +23,7 @@ export class TemplateSelector extends React.Component<IProps, IState> {
           <Select
             options={this.createOptionsFromTemplates()}
             clearable={true}
-            value={selected ? {label: selected.metadata.name, id: selected.id} : null}
+            value={selected ? {label: selected.metadata.name, value: selected.id} : null}
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/delivery/create/TemplateSelector.tsx]: @danielpeach PTAL
I think this was supposed to be ` {label: selected.metadata.name, value: selected.id}`.  My IDE was not complaining, but `tsc` was.

**Comment by jrsquared** [comment]
😍

**Comment by christopherthielen** [comment]
@jrsquared I switched `amazon/tsconfig.json` as well

File: app/scripts/modules/core/src/delivery/create/TemplateSelector.tsx (hunk 1 above)
Comment (danielpeach) [inline app/scripts/modules/core/src/delivery/create/TemplateSelector.tsx]: You're right, thanks for finding.

**Pull request merged**